project/src/main/java/com/google/sps/data/ServletLibrary.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.data;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Text;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+public class ServletLibrary {
+  // Generates a random unique identifier.
+  public static String generateUUID() {
+    return UUID.randomUUID().toString();
+  }
+
+  // Checks database to see if product set exists, if it does, returns the entity.
+  public static Entity checkProductSetExists(DatastoreService datastore, String productSetDisplayName) {
+    Filter filter = new FilterPredicate("productSetDisplayName", FilterOperator.EQUAL, productSetDisplayName);
+    Query query = new Query("ProductSet").setFilter(filter);
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    return result;
+  }
+
+  // Returns an Account object with all the information that is stored in datstore.
+  // If the account is not in datastore, returns null.
+  public static Account retrieveAccountInfo(DatastoreService datastore, UserService userService, String userId) {
+    Filter filter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    Query query = new Query("Account").setFilter(filter);
+    PreparedQuery pq = datastore.prepare(query);
+    Entity entity = pq.asSingleEntity();
+
+    if (entity == null) return null;
+
+    String logoutUrl = userService.createLogoutURL("/index.html");
+    String nickname = entity.getProperty("nickname").toString();
```

Comment:
I'm not familiar with the Entity type, but you might want to use `instanceof` to check these are Strings. It would be good to check the isUserBusinessOwner below as well.